### PR TITLE
feat: support ping (icmp) probe

### PR DESCRIPTION
**Pull request by haoel**
support the following probe configuration
```yaml
ping:
   - name: Localhost
     host: 127.0.0.1
     count: 5 # number of packets to send, default: 3
     lost: 0.2 # 20% lost percentage threshold, mark it down if the loss is greater than this, default: 0
```

### CODE REVIEW

File: docs/Manual.md [hunk 1]
```diff
@@ -61,11 +62,12 @@ EaseProbe supports the following probing methods: **HTTP**, **TCP**, **TLS**, **
...
 - **TCP**. Just simply check whether the TCP connection can be established or not. ( [TCP Probe Configuration](#72-tcp-probe-configuration) )
-- **Shell**. Run a Shell command and check the result. ( [Shell Command Probe Configuration](#73-shell-command-probe-configuration) )
-- **SSH**. Run a remote command via SSH and check the result. Support the bastion/jump server  ([SSH Command Probe Configuration](#74-ssh-command-probe-configuration))
-- **TLS**. Ping the remote endpoint, can probe for revoked or expired certificates ( [TLS Probe Configuration](#75-tls-probe-configuration) )
-- **Host**. Run an SSH command on a remote host and check the CPU, Memory, and Disk usage. ( [Host Load Probe](#76-host-resource-usage-probe-configuration) )
-- **Client**. Currently, support the following native client. Support the mTLS. ( refer to: [Native Client Probe Configuration](#77-native-client-probe-configuration) )
+- **Ping**. Just simply check whether can be ping or not. ( [Ping Probe Configuration](#73-ping-probe-configuration) )
```
Comment (proditis) [inline docs/Manual.md:65]:
```suggestion
- **Ping**. Just simply check whether can be pinged or not. ( [Ping Probe Configuration](#73-ping-probe-configuration) )
```

File: probe/ping/ping.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		return false, err.Error()
+	}
+	stats := pinger.Statistics() // get send/receive/rtt stats
+	p.ExportMetrics(stats)
+
+	stat := ""
+	stat += fmt.Sprintf("%d packets transmitted, %d packets received, %v%% packet loss\n",
+		stats.PacketsSent, stats.PacketsRecv, stats.PacketLoss)
+	stat += fmt.Sprintf("round-trip min/avg/max/stddev = %v/%v/%v/%v\n",
+		stats.MinRtt, stats.AvgRtt, stats.MaxRtt, stats.StdDevRtt)
```
Comment (proditis) [inline probe/ping/ping.go:89]: Can we maybe change these to have the DEBUG prefix? The new lines `\n` cause the messages to appear on their own and its kind of confusing. 

Unfortunately I dont know where would be the appropriate place to do such a change.

image

File: docs/Roadmap.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ The project principles, of EaseProbe' features are separated into two main categ
 
 ... with new notification backends been added constantly.
 
-## Roadmap 
+## Roadmap
 Some of the features that are planned/considered for 2022 can be broken down into three categories: *General*, *Probe*, *Notify*
```
Comment (proditis) [inline docs/Roadmap.md:57]:
```suggestion
Some of the features that we plan to implement in the future fall under one these categories: *General*, *Probe*, *Notify*
```

**Review by proditis — APPROVED** [review]
Awesome work @haoel 

I havent tested the prometheus statistics but everything else seems to be in place.

:+1: :100:

**Review by proditis — COMMENTED** [review]
Also the discord alerts seem to not be properly formatted, see attached screenshot
image

**Review by proditis — APPROVED** [review]
LGTM

**Review by samanhappy — APPROVED** [review]

**Pull request merged**